Modules/posixmodule.c

```diff
@@ -5579,11 +5714,30 @@ os_mkdir_impl(PyObject *module, path_t *path, int mode, int dir_fd)
 
 #ifdef MS_WINDOWS
     Py_BEGIN_ALLOW_THREADS
-    result = CreateDirectoryW(path->wide, NULL);
+    switch (mode) {
+    case 0x1C0: // 0o700
+        error = initializeMkdir700SecurityAttributes(&pSecAttr, &secAttrData);
+        break;
+    default:
+        error = initializeDefaultSecurityAttributes(&pSecAttr, &secAttrData);
+        break;
+    }
+    if (!error) {
+        result = CreateDirectoryW(path->wide, pSecAttr);
+        error = clearSecurityAttributes(&pSecAttr, &secAttrData);
+    } else {
+        // Ignore errors - we have a more interesting one already
+        clearSecurityAttributes(&pSecAttr, &secAttrData);
```

Comment:
Doesn't seem like this ignores the error, line 5734 will still see it.

Reply:
It ignores the error from the subsequent `clearSecurityAttributes` call (unlike on 5727) - I'll clarify the comment.

Reply:
Ah, I see what you mean. If possible, it would also be good to have tests that exercise this error path.